### PR TITLE
Add possibility to test preference prediction

### LINKED ISSUE #45: Add test set in preference classification
Add option to specify if a test set (where the performance of the preference classifier is measured) is used in the `DomainScoring` class.

- [ ] Separate tests examples in `_extract_training_data_labels` method
- [ ] Do we also need a validation set?

### PR DESCRIPTION
**Pull request by Baschdl**
Closes #45

### CODE REVIEW

File: domain_scoring/domain_scoring.py [hunk 1]
```diff
@@ -17,19 +18,30 @@ def __init__(self):
         """
         # The token_pattern also allows single character strings which the default doesn't allow
         self.vectorizer = TfidfVectorizer(analyzer='word', token_pattern='\\b\\w+\\b')
-        self.classifier = DecisionTreeClassifier()
+        self.classifier = DecisionTreeClassifier(random_state=42)
```
Comment (GittiHab) [inline domain_scoring/domain_scoring.py]: Very random  :stuck_out_tongue_winking_eye:

File: domain_scoring/domain_scoring.py [hunk 2]
```diff
@@ -17,19 +18,30 @@ def __init__(self):
...
         self.domain_value_transformer = NaiveTransformer()
 
-    def fit(self, metapath_graph: MetaPathRatingGraph) -> None:
+    def fit(self, metapath_graph: MetaPathRatingGraph, test_size: float = False) -> None:
```
Comment (GittiHab) [inline domain_scoring/domain_scoring.py]: I guess `None` is a more intuitive value than `False` in this case.

File: domain_scoring/domain_scoring.py [hunk 3]
```diff
@@ -17,19 +18,30 @@ def __init__(self):
...
         """
         Fits a classifier to predict a meta-path ordering.
+        :param test_size: Specify size of test set if a test accuracy should be reported
```
Comment (GittiHab) [inline domain_scoring/domain_scoring.py]: I would put this after `metapath_graph` and mention that if left away than no accuracy is reported.

**Comment by Baschdl** [comment]
The non-random setting of the random state of the decision tree was on purpose so that we produce reproducable results. The addition you made is redundant to just not specifying the random state in the constructor. (Sorry, can't comment properly on old iPad)

**Comment by GittiHab** [comment]
I don't like the idea of hardcoding reproduceability. This should somehow be added to the config.

**Comment by Baschdl** [comment]
What would you specify in the config? If each run of our system yields the system result or sth. like the random state?

**Comment by GittiHab** [comment]
I could imagine either one of them – or both.

**Comment by Baschdl** [comment]
What's your use case for randomized results? Maybe a better solution is 42 as the standard setting of a parameter.

**Comment by Baschdl** [comment]
@GittiHab I would like to finish this PR. Could you please elaborate a bit more on your use case?

**Comment by GittiHab** [comment]
I added the random state to the config file. You can have a look at it – of course I am biased if I approve this PR ;)

**Comment by Baschdl** [comment]
I like it

**Pull request merged**